src/utils/command.ts

```diff
@@ -11,29 +11,41 @@ const mentionToRegex: MentionToRegex = {
   mentionChannel: /<#(\d{17,19})>/
 }
 
-export type CommandArgumentMatchTypes =
-  | 'flag'
-  | 'mentionUser'
-  | 'mentionRole'
-  | 'mentionChannel'
-  | 'content'
-  | 'rest'
-
-export interface Args<T = unknown> {
+interface ArgsBase {
   name: string
-  match: CommandArgumentMatchTypes
-  defaultValue?: T
-  flag?: string
 }
 
+interface Flag extends ArgsBase {
```

Comment:
Names like `Flag`, `Mention`, etc are too short and not so specific about what they are. Append `Arg` to them? Also, export them for docs.

Reply:
I wasn't sure if these needed to be exported because they are components of the type `Args` but I can export those